### PR TITLE
Add support for custom timestamp. Closes #103

### LINKED ISSUE #103: UTC timezone
Is there any option to log with `UTC` instead of the location time zone?

I found a `time.Now()` in https://github.com/hashicorp/go-hclog/blob/master/intlogger.go#L155, but couldn't find anything else

### PR DESCRIPTION
**Pull request by binaek**

### CODE REVIEW

**Comment by evanphx** [comment]
Is there a reason you'd want the logs in a different timezone than the system itself? That's the reason we don't do this currently, we want the logs to always reflect the time of the system.

**Comment by binaek** [comment]
@evanphx  I put in the `UTC` behind a flag for precisely the reason you mentioned. Most logs need to reflect the time of the system where it's logged from.

Having said that, `UTC` as a time zone removes a lot of ambiguities associated with local time zones and timestamps (not even bringing DST into the picture)

However, the motivation behind this PR is that we use `hclog` inside a library - which runs in an application which has its own logging and can be setup to log in UTC. The logs from our library interleave with the application's log and appear in the same log destination.

**Comment by evanphx** [comment]
@binaek Gotcha, that makes sense. I'm wondering if rather than just a flag to set UTC, we should allow for someone to pass in a function that returns the current time. That would allow them maximum flexibility to calculate the current time in whatever form they want.

**Comment by binaek** [comment]
@evanphx That sounds like a good approach. Let me have a dig and update the PR.

**Comment by binaek** [comment]
@evanphx I have updated the PR with support for a custom time function - with a default which returns `time.Now()`

Take a look when you get the chance.

File: global.go [hunk 1]
```diff
@@ -14,6 +15,7 @@ var (
 	DefaultOptions = &LoggerOptions{
 		Level:  DefaultLevel,
 		Output: DefaultOutput,
+		TimeFn: func() time.Time { return time.Now() },
```
Comment (evanphx) [inline global.go:18]:
```suggestion
		TimeFn: time.Now,
```

File: intlogger.go [hunk 2]
```diff
@@ -117,7 +117,7 @@ func newLogger(opts *LoggerOptions) *intLogger {
 		json:              opts.JSONFormat,
 		name:              opts.Name,
 		timeFormat:        TimeFormat,
-		useUtcTime:        opts.UTCTime,
+		timeFn:            func() time.Time { return time.Now() },
```
Comment (evanphx) [inline intlogger.go:120]:
```suggestion
		timeFn:            time.Now,
```

**Review by evanphx — CHANGES_REQUESTED** [review]
Super minor bits, no need to create a closure here, can just use `time.Now` directly.

**Comment by binaek** [comment]
Huh! Not sure why I put the closure in there. Let me update the PR when I am at my desk

**Comment by binaek** [comment]
@evanphx PR updated

**Comment by binaek** [comment]
@evanphx let me know if you think this requires other changes

**Pull request merged**